logstash-core/src/main/java/org/logstash/config/ir/compiler/OutputStrategyExt.java

```diff
@@ -248,11 +252,12 @@ protected SimpleAbstractOutputStrategyExt(final Ruby runtime, final RubyClass me
         @JRubyMethod(required = 4)
         public IRubyObject initialize(final ThreadContext context, final IRubyObject[] args) {
             final RubyClass outputClass = (RubyClass) args[0];
+            final ExecutionContextExt executionContext = (ExecutionContextExt) args[2];
+            // TODO: fixup mocks
             // Calling "new" here manually to allow mocking the ctor in RSpec Tests
-            output = args[0].callMethod(context, "new", args[3]);
+            output = PLUGIN_CONTEXTUALIZER_MODULE.callMethod("initialize_plugin", executionContext, outputClass, args[3]);
```

Comment:
I can't find a way to define the factory on `Plugin` and have it available to the java unit tests (especially since those manually create "mock" inputs/outputs that do not inherit from `LogStash::Plugin`).